pkg/trace/pb/decoder_test.go

```diff
@@ -46,24 +46,56 @@ func TestParseFloat64(t *testing.T) {
 	assert.Equal(3.14, f)
 }
 
-func TestDecode(t *testing.T) {
+func TestParseFloat64Bytes(t *testing.T) {
+	assert := assert.New(t)
+
+	data := []byte{
+		0x2a,             // 42
+		0xd1, 0xfb, 0x2e, // -1234
+		0xcd, 0x0a, 0x9b, // 2715
+		0xcb, 0x40, 0x09, 0x1e, 0xb8, 0x51, 0xeb, 0x85, 0x1f, // float64(3.14)
+	}
+
+	var f float64
+	var err error
+	bts := data
+
+	f, bts, err = parseFloat64Bytes(bts)
+	assert.NoError(err)
+	assert.Equal(42.0, f)
+
+	f, bts, err = parseFloat64Bytes(bts)
+	assert.NoError(err)
+	assert.Equal(-1234.0, f)
+
+	f, bts, err = parseFloat64Bytes(bts)
+	assert.NoError(err)
+	assert.Equal(2715.0, f)
+
+	f, _, err = parseFloat64Bytes(bts)
+	assert.NoError(err)
+	assert.Equal(3.14, f)
+}
+
+func TestDecodeBytes(t *testing.T) {
 	want := Traces{
 		{{Service: "A", Name: "op"}},
 		{{Service: "B"}},
 		{{Service: "C"}},
 	}
-	var buf bytes.Buffer
-	if err := msgp.Encode(&buf, &want); err != nil {
+	var bts []byte
```

Comment:
Note that I usually tended to group multiple variable definitions using `var ( ...` as opposed to C-like `var` statements. No biggie, but it would be nice to try and keep it consistent. Arguably I know that there's some old code somewhere which goes against my statement here.